pkg/k8s/watchers/pod.go

```diff
@@ -263,6 +263,12 @@ func (k *K8sPodWatcher) podsInit(asyncControllers *sync.WaitGroup) {
 		return
 	}
 
+	// If CiliumEndpointCRD is disabled and network policies are disabled, watch local pods.
+	if !option.NetworkPolicyEnabled(option.Config) {
```

Comment:
nit: This condition can be merged with the conditions above, since they have the same actions.

Consider using `cmp.Or()` to combine the conditions.

Reply:
Thanks for the suggestion, updated based on this :)